scene/gui/popup_button.h

```diff
@@ -0,0 +1,63 @@
+/*************************************************************************/
+/*  popup_button.h                                                       */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2022 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2022 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef POPUP_BUTTON_H
+#define POPUP_BUTTON_H
+
+#include "scene/gui/button.h"
+#include "scene/gui/popup.h"
+
+class PopupButton : public Button {
```

Comment:
If you want to do this, then you should go all the way and handle OptionButton and ColorPicker, but I am not entirely sure this is justified.

Reply:
What would you suggest? Users would expect the button to toggle on hover, just like MenuButton does. I could not do this, but then we'll likely get complaints that this behavior doesn't happen.

I don't think that OptionButton fits here, but ColorPickerButton for sure could be updated to use this new base node. But I'd prefer to do that in a separate PR, because this one doesn't depend on ColorPickerButton, and it doesn't touch what it doesn't need to touch, preserving compatibility.